params/config.go

```diff
@@ -299,8 +318,9 @@ type ChainConfig struct {
 	TerminalTotalDifficultyPassed bool `json:"terminalTotalDifficultyPassed,omitempty"`
 
 	// Various consensus engines
-	Ethash *EthashConfig `json:"ethash,omitempty"`
-	Clique *CliqueConfig `json:"clique,omitempty"`
+	Ethash *EthashConfig        `json:"ethash,omitempty"`
+	Clique *CliqueConfig        `json:"clique,omitempty"`
+	Dev    *DeveloperModeConfig `json:"dev,omitempty"`
```

Comment:
Just my 2 cents here: I feel this is overkill, because you can just pass it as a command line option, and use that parameter when instantiating the `CLMock`. So you could reduce the footprint of that PR by not adding more config here, since it's really only used in one place.

Reply:
Yes but it must be persisted in the case where dev mode is not being used ephemerally.  Hence I add `DeveloperModeConfig` so that the period will be stored.

Reply:
When you specify `--datadir` and `--dev` together it persists the chain.  if the datadir is omitted, the chain is ephemeral.

Reply:
I also feel this is overkill? Is there ever really a use case to have a dev mode genesis? `ChainConfig` is used quite a bit and so we should be confident in the addition here.

Reply:
In `DeveloperModeConfig `, there is only a single field `period`. It's totally fine to use another value in the next restart I think. We can have a default value for it, e.g. 12s or so, and users can change it by CLI flag.